packages/rocketchat-lib/server/functions/deleteUser.js

```diff
@@ -19,12 +19,13 @@ RocketChat.deleteUser = function(userId) {
 		RocketChat.models.Subscriptions.db.findByUserId(userId).forEach((subscription) => {
```

Comment:
I think this code is wrong, we dont delete orphan subscriptions and we dont test if the user is the last owner/moderator and etc....

Reply:
RocketChat.models.Subscriptions.db.findByUserId(userId).forEach(({rid}) => {
            const room = RocketChat.models.Rooms.findOneById(rid);
            RocketChat.models.Subscriptions.removeByRoomId(rid);
			if (room) {
                if (room.t === 'd') {
                    RocketChat.models.Messages.removeByRoomId(rid);
					return RocketChat.models.Rooms.removeById(rid);
                }
                if (room.usernames && room.usernames.length === 1) {
                  RocketChat.models.Rooms.removeById(rid); // Remove non-channel rooms with only 1 user (the one being deleted)
                }
			}
		});